feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/vote/RealGovernanceLocksEstimator.kt

```diff
@@ -0,0 +1,155 @@
+package io.novafoundation.nova.feature_governance_impl.domain.referendum.vote
+
+import io.novafoundation.nova.common.data.network.runtime.binding.BlockNumber
+import io.novafoundation.nova.common.utils.orZero
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.AccountVote
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.OnChainReferendum
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.ReferendumId
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.TrackId
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.TrackInfo
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.Voting
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.amount
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.asOngoingOrNull
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.flattenCastingVotes
+import io.novafoundation.nova.feature_governance_api.domain.locks.RealClaimScheduleCalculator
+import io.novafoundation.nova.feature_governance_api.domain.locks.reusable.LocksChange
+import io.novafoundation.nova.feature_governance_api.domain.locks.reusable.ReusableLock
+import io.novafoundation.nova.feature_governance_api.domain.locks.reusable.addIfPositive
+import io.novafoundation.nova.feature_governance_api.domain.locks.reusable.maximize
+import io.novafoundation.nova.feature_governance_api.domain.referendum.common.Change
+import io.novafoundation.nova.feature_governance_api.domain.referendum.vote.GovernanceVoteAssistant
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.types.Balance
+import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
+import io.novafoundation.nova.feature_wallet_api.domain.model.BalanceLock
+import io.novafoundation.nova.feature_wallet_api.domain.model.maxLockReplacing
+import io.novafoundation.nova.feature_wallet_api.domain.model.transferableReplacingFrozen
+import io.novafoundation.nova.runtime.util.BlockDurationEstimator
+
+internal class RealGovernanceLocksEstimator(
```

Comment:
Tbh would be good to see a proper diff with what we had before since it is a pretty big class to review but I will try my best